src/main/webapp/resources/js/pages/projects/settings/components/SettingsNav.jsx

```diff
@@ -1,61 +1,105 @@
-import { Link } from "@reach/router";
 import { Menu } from "antd";
 import React from "react";
+import { Link, useLocation } from "react-router-dom";
 
 /**
  * Component to handle navigation within the project settings page
- * @param {string} path - the current page name
  * @returns {JSX.Element}
  * @constructor
  */
 export default function SettingsNav({
-  path,
-  showRemote = false,
-  canManage = false,
-}) {
-  const [key, setKey] = React.useState();
+                                      basePath,
+                                      showRemote = false,
+                                      canManage = false
+                                    }) {
+                                      const location = useLocation();
+                                      const [key, setKey] = React.useState();
 
-  React.useEffect(() => {
-    setKey(path.split("/")[0]);
-  }, [path]);
+                                      const keyRegex =
+                                        /\/projects\/\d+\/settings\/([\w_-]+)/;
+                                      React.useEffect(() => {
+                                        const path = keyRegex.exec(
+                                          location.pathname
+                                        );
+                                        if (path) {
+                                          setKey(path[1]);
+                                        } else {
+                                          throw new Error(
+                                            "Developer: You ed the schema for the settings URL - fix this"
```

Comment:
Might want to change this error message :)

Reply:
Removed in 123c9e84c9a6ccadd9169144cdc7a975f16b69b6